src/Analyzers/CSharp/Analyzers/UseNullPropagation/CSharpUseNullPropagationDiagnosticAnalyzer.cs

```diff
@@ -36,29 +41,66 @@ protected override bool IsInExpressionTree(SemanticModel semanticModel, SyntaxNo
             => node.IsInExpressionTree(semanticModel, expressionTypeOpt, cancellationToken);
 
         protected override bool TryAnalyzePatternCondition(
-            ISyntaxFacts syntaxFacts, SyntaxNode conditionNode,
-            [NotNullWhen(true)] out SyntaxNode? conditionPartToCheck, out bool isEquals)
+            ISyntaxFacts syntaxFacts, ExpressionSyntax conditionNode,
+            [NotNullWhen(true)] out ExpressionSyntax? conditionPartToCheck, out bool isEquals)
         {
             conditionPartToCheck = null;
             isEquals = true;
 
             if (conditionNode is not IsPatternExpressionSyntax patternExpression)
-            {
                 return false;
-            }
 
-            if (patternExpression.Pattern is not ConstantPatternSyntax constantPattern)
+            var pattern = patternExpression.Pattern;
+            if (pattern is UnaryPatternSyntax(SyntaxKind.NotPattern) notPattern)
```

Comment:
this is new functionality for the existing analyzer.  we now detect `x is not null ? x.Y : null` which we didn't detect before.